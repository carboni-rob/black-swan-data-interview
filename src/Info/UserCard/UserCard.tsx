import React from "react";

type Props = {
  user: any;
};

const UserCard = ({ user }: Props): JSX.Element => {
  return (
    <div className="card">
      <h2>User Info:</h2>
      <img src={user.avatar_url} alt="User Avatar" className="userAvatar" />
      <h3>User name: {user.name || "not set"}</h3>
      <h3>Company: {user.company || "not set"}</h3>
      <h3>Email: {user.email || "not set"}</h3>
      <h3>Followers: {user.followers}</h3>
      <h3>Following: {user.following}</h3>
    </div>
  );
};

export default UserCard;
