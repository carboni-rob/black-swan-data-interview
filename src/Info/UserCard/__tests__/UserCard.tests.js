/* eslint-disable @typescript-eslint/camelcase */
import React from "react";
import { render } from "@testing-library/react";
import UserCard from "../UserCard";

describe("<UserCard />", () => {
  const user = {
    name: "Roberto",
    company: "Black Swan Data",
    email: "carboni@robertocarboni.it",
    followers: 100,
    following: 100,
    avatar_url: "fake/avatar/path"
  };
  const tree = <UserCard {...user} />;

  it("should match last snapshot", () => {
    const { container } = render(tree);

    expect(container.firstChild).toMatchSnapshot();
  });

  it("should contain a title", () => {
    const { getByText } = render(tree);

    getByText("User Info:");
  });

  it("should contain an image with the correct src and class", () => {
    const { getByAltText } = render(tree);

    const avatar = getByAltText("User Avatar");

    expect(avatar).toHaveProperty("className", "userAvatar");
    expect(avatar).toHaveProperty("src", "http://localhost/fake/avatar/path");
  });

  it("should contain a user name field with the appropriate text", () => {
    const { getByText } = render(tree);

    getByText("User name: Roberto");
  });

  it("should contain a user name field with 'not set' text when username is missing", () => {
    const { getByText } = render(<UserCard {...user} name="" />);

    getByText("User name: not set");
  });

  it("should contain a company field with the appropriate text", () => {
    const { getByText } = render(tree);

    getByText("Company: Black Swan Data");
  });

  it("should contain a company field with 'not set' text when company name is missing", () => {
    const { getByText } = render(<UserCard {...user} company="" />);

    getByText("Company: not set");
  });

  it("should contain an email field with the appropriate text", () => {
    const { getByText } = render(tree);

    getByText("Email: carboni@robertocarboni.it");
  });

  it("should contain an email field with 'not set' text when email is missing", () => {
    const { getByText } = render(<UserCard {...user} email="" />);

    getByText("Email: not set");
  });

  it("should contain a followers field with the appropriate followers count", () => {
    const { getByText } = render(tree);

    getByText("Followers: 100");
  });

  it("should contain a following field with the appropriate following count", () => {
    const { getByText } = render(tree);

    getByText("Following: 100");
  });
});
