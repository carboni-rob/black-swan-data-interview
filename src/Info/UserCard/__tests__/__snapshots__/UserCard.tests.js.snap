// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserCard /> should match last snapshot 1`] = `
<div
  class="card"
>
  <h2>
    User Info:
  </h2>
  <img
    alt="User Avatar"
    class="userAvatar"
    src="fake/avatar/path"
  />
  <h3>
    User name: 
    Roberto
  </h3>
  <h3>
    Company: 
    Black Swan Data
  </h3>
  <h3>
    Email: 
    carboni@robertocarboni.it
  </h3>
  <h3>
    Followers: 
    100
  </h3>
  <h3>
    Following: 
    100
  </h3>
</div>
`;
